components/client/src/main/java/com/hotels/styx/client/StyxHttpClient.java

```diff
@@ -336,19 +335,27 @@ public String toString() {
      * A builder for {@link com.hotels.styx.client.StyxHttpClient}.
      */
     public static class Builder {
-        private final BackendService backendService;
+
+        private final Id backendServiceId;
         private MetricRegistry metricsRegistry = new CodaHaleMetricRegistry();
         private List<RewriteRule> rewriteRules = emptyList();
         private RetryPolicy retryPolicy = new RetryNTimes(3);
         private LoadBalancer loadBalancer;
         private boolean contentValidation;
         private OriginStatsFactory originStatsFactory;
         private String originsRestrictionCookieName;
+        private StickySessionConfig stickySessionConfig = stickySessionDisabled();
 
-        public Builder(BackendService backendService) {
-            this.backendService = checkNotNull(backendService);
+        public Builder(Id backendServiceId) {
+            this.backendServiceId = checkNotNull(backendServiceId);
```

Comment:
Please use Java 8's `requireNonNull` instead of Guava's `checkNotNull`. Same comment for `stickySessionConfig(...)`

Reply:
See ticket https://github.com/HotelsDotCom/styx/issues/243